File: C/PrimeX.c
#include <stdio.h>
#include <stdlib.h>

#define NL 1000000000   // number limit: 1 billion
#define LL 11           // line limit: 11 characters

int main()
{
    printf("========\n PrimeX\n========\n\n");

    FILE *fp = fopen("primes.txt", "wb");
    
    const int N = NL + 3;
    char* primes = (char*)calloc(N, sizeof(char));
    char* line = (char*)calloc(LL+1, sizeof(char));
    int l = 0;

    for (int i = 2; i < 3; ++i)
    {
        switch (primes[i])
        {
            case 0:
            itoa(i, line, 10);
            
            while (line[l] != '\0')
            {
                ++l;
            }
            line[l] = '\n';
            fwrite(line, sizeof(char), l+1, fp);

            for (int j = i+i; j < N; j += i)
            {
                primes[j] = 1;
            }
        }
    }

    for (int i = 3; i < N; i += 2)
    {
        switch (primes[i])
        {
            case 0:
            itoa(i, line, 10);
            
            while (line[l] != '\0')
            {
                ++l;
            }
            line[l] = '\n';
            fwrite(line, sizeof(char), l+1, fp);

            for (int j = i+i; j < N; j += i)
            {
                primes[j] = 1;
            }
        }
    }

    free(primes);
    free(line);
    fclose(fp);

    printf("Done :D\nCheck your folder for the results");

    return 0;
}

File: Java/PrimeX.java
package Java;

import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;

public class PrimeX {
    public static final int NL = 1000000000;

    public static void main(String[] args) throws IOException {
        System.out.println("========\n PrimeX\n========\n");

        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter("primes.txt"));

        final int N = NL + 2;
        boolean[] primes = new boolean[N];

        for (int i = 2; i < 3; ++i) {
            if (!primes[i]) {
                bufferedWriter.write("" + i);
                bufferedWriter.newLine();

                for (int j = i; j < N; j += i) {
                    primes[j] = true;
                }
            }
        }

        for (int i = 3; i < N; i += 2) {
            if (!primes[i]) {
                bufferedWriter.write("" + i);
                bufferedWriter.newLine();

                for (int j = i; j < N; j += i) {
                    primes[j] = true;
                }
            }
        }

        bufferedWriter.close();

        System.out.println("Done :D\nCheck your folder for the results\n");
    }
}